deployability/modules/testing/testing.py

```diff
@@ -15,29 +14,39 @@ class Tester:
     def run(cls, payload: InputPayload) -> None:
         payload = InputPayload(**dict(payload))
         inventory = Inventory(**Utils.load_from_yaml(payload.inventory))
-        extra_vars = cls._get_extra_vars(payload)
+        extra_vars = cls._get_extra_vars(payload).model_dump()
+        dependencies_dict = {}
+        for dependency in extra_vars['dependencies']:
+            dependency = ast.literal_eval(dependency)
+            dependencies_dict.update(dependency)
+        extra_vars['dependencies'] = dependencies_dict
```

Comment:
It would be better if `extra_vars` keeps being an `ExtraVars` object for the validations provided by pydantic, this dependencies dict could be generated inside `_get_extra_vars()` (also, the dependencies attribute of ExtraVars could be modified to be a dict)
```suggestion
        extra_vars = cls._get_extra_vars(payload)
```
the lines 24 and 27 should also have the dot notation

Reply:
The problem of changing this structure is that when the test is executed from Workflow:

```
--dependencies '{"manager": "/tmp/dtt1-poc/manager-linux-ubuntu-22.04-amd64/inventory.yaml"}','{"agent": "/tmp/dtt1-poc/agent-linux-ubuntu-22.04-amd64/inventory.yaml"}' 
```

This structure has a list of 2 small dictionaries, but over that, it is a list with 2 dictionaries inside.

As the input is a list of dictionaries, the InputPayload filtered it saying that the structure is not a dictionary, it is a list.

Then the idea was to design taking those 2 dicts as strings and changing them before running the test.

On the other hand, if I make those changes from _get_extra_vars, changing it for a dictionary, the second validation have said that the dictionary is not a list.